protocols/v2/subprotocols/template-distribution/src/request_transaction_data.rs

```diff
@@ -58,6 +66,38 @@ pub struct RequestTransactionDataSuccess<'decoder> {
     pub transaction_list: Seq064K<'decoder, B016M<'decoder>>,
 }
 
+/// Message used by an upstream to respond successfully to a [`RequestTransactionData`] message.
+///
+/// A response to [`RequestTransactionData`] which contains the set of full transaction data and
+/// excess data required for block validation. For practical purposes, the excess data is usually
+/// the SegWit commitment, however the Job Declarator **must not** have any assumptions about it.
+///
+/// Note that the transaction data **must** be treated as opaque blobs and **must** include any
+/// SegWit or other data which the downstream may require to verify the transaction. For practical
+/// purposes, the transaction data is likely the witness-encoded transaction today. However, to
+/// ensure backward compatibility, the transaction data **may** be encoded in a way that is
+/// different from the consensus serialization of Bitcoin transactions.
+///
+/// The [`RequestTransactionDataSuccess`] sender **must** ensure that provided data is forward and
+/// backward compatible. This way the receiver of the data can interpret it, even in the face of
+/// new, consensus-optional data.  This allows significantly more flexibility on both the
+/// [`RequestTransactionDataSuccess`] generating and interpreting sides during upgrades, at the
+/// cost of breaking some potential optimizations which would require version negotiation to
+/// provide support for previous versions.
+///
+/// Having some method of negotiating the specific format of transactions between the Template
+/// Provider and the downstream would be helpful but overly burdensome, thus the above requirements
+/// are made explicit.
+///
+/// As a result, and as a non-normative suggested implementation for Bitcoin Core, this implies
+/// that additional consensus-optional data appended at the end of transaction data will simply be
+/// ignored by versions which do not understand it.
+///
+/// To work around the limitation of not being able to negotiate e.g. a transaction compression
+/// scheme, the format of the opaque data in [`RequestTransactionDataSuccess`] messages **may** be
+/// changed in a non-compatible way at the time of fork activation, given sufficient time from
+/// code-release to activation and there being in protocol(Template Declaration) signaling of
+/// support for the new fork (e.g. for soft-forks activated using [BIP 9]).
```

Comment:
Can we just remove this description, and just add C version of RequestTransactionDataSuccess..

Reply:
Same, this is for refactoring not documentation